server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -893,24 +900,40 @@ private void deleteTablets(MergeInfo info, Range scanRange, BatchWriter bw, Accu
     Manager.log.debug("Deleting range {}", scanRange);
     scanner.setRange(scanRange);
     RowIterator rowIter = new RowIterator(scanner);
+    KeyExtent lastTablet = null;
     while (rowIter.hasNext()) {
       Iterator<Entry<Key,Value>> row = rowIter.next();
       m = null;
       while (row.hasNext()) {
         Entry<Key,Value> entry = row.next();
         Key key = entry.getKey();
+        lastTablet = KeyExtent.fromMetaRow(key.getRow());
 
         if (m == null) {
           m = new Mutation(key.getRow());
         }
 
-        m.putDelete(key.getColumnFamily(), key.getColumnQualifier());
-        Manager.log.debug("deleting entry {}", key);
+        // For delete operations, check if this is the last tablet
+        // If this is the last tablet in the table then only delete
+        // the files as we need to make sure we always keep at least 1 tablet
+        if (info.isDelete() && lastTablet.endRow() == null) {
+          if (key.getColumnFamily().equals(DataFileColumnFamily.NAME)) {
+            m.putDelete(key.getColumnFamily(), key.getColumnQualifier());
+            Manager.log.debug("deleting file entry {}", key);
+          }
+        } else {
+          m.putDelete(key.getColumnFamily(), key.getColumnQualifier());
+          Manager.log.debug("deleting entry {}", key);
+        }
+      }
+      if (m != null && !m.getUpdates().isEmpty()) {
```

Comment:
This is an interesting case, I suppose before the mutation always had something in it.  Now if last tablet has no files the mutation will be empty.